src/box/lua/schema.lua

```diff
@@ -328,10 +330,23 @@ local function feedback_save_event(event)
     end
 end
 
-box.begin = function()
+box.begin = function(options)
+    if options then
+        check_param(options, 'options', 'table')
+    end
+    local timeout = options and options["timeout"] or box.cfg.txn_timeout
+    if type(timeout) ~= "number" then
+        box.error(box.error.ILLEGAL_PARAMS, "transaction timeout " ..
+                  "should be of type number")
+    end
+    if timeout <= 0 then
+        box.error(box.error.ILLEGAL_PARAMS, "transaction timeout " ..
+                  "must be greather than 0")
+    end
```

Comment:
Same comment as for stream_begin.

Reply:
fixed, but i still use box.error(box.error.ILLEGAL_PARAMS, because in this file we always use it, not 'error'